keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/CombinedRepository.kt

```diff
@@ -81,26 +81,31 @@ class CombinedRepository(
 
   @Transactional(propagation = REQUIRED)
   override fun upsertDeliveryConfig(deliveryConfig: DeliveryConfig): DeliveryConfig {
-    val old = try {
+    val configWithSameName = try {
```

Comment:
Renamed some variables, let me know if it makes sense..

Reply:
The names are a bit long, but definitely more self-explanatory than the previous ones, so I think they're ok. The only thought that occurred to me is whether we might be over-complicating things unnecessarily given the constraint we want to enforce. Specifically, since each app can only have one delivery config, what's the point of having the user specify a name for the delivery config in the first place? We could just index on the app name... But the problem with changing that is that we'd need to go around and fix everyone's delivery configs, so I'm discouraged for that reason...

Reply:
@luispollo One thing we could do is make the delivery config name optional (under the hood we could just default the delivery config name to the app name), and "deprecate" the use of a delivery config name. This would avoid breaking existing users.

Reply:
Yeah, I see what you are saying. Fwiw, I think 99% of our users will not even specify the delivery config name, so I believe we won't really hit issues regarding config names anymore. I hope this is the last fix in that space!

Reply:
@lorin delivery config names are optional, and they are defaulting to "appName-manifest".

Reply:
@gal-yardeni I didn't realize that! I think we should eventually stop letting people give their delivery configs custom names.

Reply:
@lorin I'm on the fence about it :) we can raise this question to the team.

Reply:
The problem with that would be that it bakes in the one-config-per-app constraint that we're not sure we want in the long term.